MovieMVVM/MovieMVVM/Presentation/Movies/ViewModel/MoviesViewModel.swift

```diff
@@ -51,22 +62,27 @@ final class MoviesViewModel: MoviesViewModelProtocol {
         isLoading = true
         moviesViewData?(.loading)
         page = pagination ? page + Constants.one : page
-        networkService.fetchMovies(kind: kind, page: page) { [weak self] result in
-            guard let self else { return }
-            self.isLoading = false
-            DispatchQueue.main.async { [weak self] in
+        dataService
+            .fetchMovies(
+                kind: kind,
+                page: page,
+                apiKey: storageKeyChain.readValueFromKeyChain(from: .apiKey)
+            ) { [weak self] result in
                 guard let self else { return }
-                switch result {
-                case let .success(response):
-                    self.page = response.page
-                    self.totalPages = response.totalPages
-                    self.movies = pagination ? self.movies + response.movies : response.movies
-                    self.moviesViewData?(.success)
-                case .failure:
-                    self.moviesViewData?(.failure)
+                self.isLoading = false
+                DispatchQueue.main.async { [weak self] in
```

Comment:
```suggestion
                DispatchQueue.main.async {  in
```
тут не должно быть захвата